spec/loophp/collection/Utils/CallbacksArrayReducerSpec.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\loophp\collection\Utils;
+
+use ArrayIterator;
+use loophp\collection\Utils\CallbacksArrayReducer;
+use PhpSpec\ObjectBehavior;
+
+class CallbacksArrayReducerSpec extends ObjectBehavior
+{
+    public function it_is_initializable()
+    {
+        $this->shouldHaveType(CallbacksArrayReducer::class);
+    }
+
+    public function it_reduce_callbacks(): void
+    {
+        $callbacks = [
+            static fn (int $v): bool => 5 < $v,
+            static fn (int $v): bool => 0 === $v % 2,
+        ];
+
+        $input = range(0, 10);
+
+        $iterator = new ArrayIterator($input);
+
+        $this::or()($callbacks, 0, 0, $iterator)
+            ->shouldReturn(true);
+
+        $this::or()($callbacks, 13, 0, $iterator)
+            ->shouldReturn(true);
+
+        $this::or()($callbacks, 3, 0, $iterator)
+            ->shouldReturn(false);
+    }
```

Comment:
A few comments here:
- the `$iterator` is not really used in this test because none of the callbacks use it. Would be good to have one test which uses it
- similarly, would be good to have a test with callbacks that use the second argument (the key)
- we should also test some edge cases, e.g. empty callbacks array or a single callback
- it's good practice to have one assertion per test. I know with `CollectionSpec` maybe that's not too convenient because it's a big class but with something like this I think it makes it easier to follow, e.g.
`it_reduces_empty_callbacks_array`
`it_reduces_single_callback`
`it_reduces_callbacks_with_key_and_value`
`it_reduces_callbacks_with_all_parameters`

Reply:
Ok, will add this tonight !

Reply:
Job done!